scripts/publish.sh

```diff
@@ -5,8 +5,8 @@ set -e
 set -o pipefail
 
 if [[ -z $NPM_TOKEN ]]; then
-  echo "Notice: NPM_TOKEN environment variable not set. Running 'npm publish --dry-run'."
-  npm publish --dry-run
+  echo "Notice: NPM_TOKEN environment variable not set. Running 'yarn pack --dry-run'."
+  yarn pack --dry-run
```

Comment:
Why not run `yarn npm publish --dry-run`?

Reply:
`yarn npm publish` doesn't have any `--dry-run` flag. This is the equivalent command; it packs the package and prints it out, just like `npm publish --dry-run` does.

Reply:
Hmm, it actually is not equivalent. It has the list of files, but not this:

```
npm notice === Tarball Details === 
npm notice name:          @metamask/controllers                   
npm notice version:       33.0.1                                  
npm notice package size:  386.8 kB                                
npm notice unpacked size: 1.8 MB                                  
npm notice shasum:        22f9a6fd5f666c13f004b83b3513a09b79cdab0c
npm notice integrity:     sha512-G/xGxxfmRkRkY[...]jy5TtHNuo2aew==
npm notice total files:   193   
```

Reply:
I compared the output to the `npm pack` command. `npm pack --dry-run` has the same output as `npm publish --dry-run`. So `yarn pack` is the right thing to use here, it's just that it has less informative output than npm.